src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -531,11 +531,12 @@ void WindowImplX11::setIcon(unsigned int width, unsigned int height, const Uint8
 
     xcb_generic_error_t* errptr = xcb_request_check(m_connection, cookie);
     bool setWindowIconFailed = (errptr != NULL);
+    uint8_t error_code = errptr->error_code;
```

Comment:
Having just glanced at the doc for `xcb_request_check` I'm wondering if the following would be better (at least clearer IMO):

```cpp
xcb_generic_error_t* errptr = xcb_request_check(m_connection, cookie);
if (errptr)
{
    err() << "Failed to set the window's icon: Error code " << errptr->error_code << std::endl;
    free(errptr);
    return;
}
```